### ISSUE TITLE
Unable to inject ModuleConfig without using name in 0.7.7

### ISSUE THREAD

**Issue by Javaman44** [Urigo/graphql-modules#550]
I was unable to inject module-configuration using last version/documentation (see: https://graphql-modules.com/docs/introduction/configuration)

```json
My dependencies:
       "dependencies": {
      "@graphql-modules/core": "^0.7.7",
      "@graphql-modules/di": "^0.7.7",
      "@types/express": "^4.17.0",
      "@types/express-graphql": "^0.8.0",
      "apollo-server": "^2.6.7",
      "deepmerge": "^3.3.0",
      "graphql": "^14.4.2",
      "graphql-import-node": "0.0.4",
      "reflect-metadata": "^0.1.13",
      "winston": "^3.2.1"
   },
   "devDependencies": {
      "@graphql-codegen/cli": "^1.3.1",
      "@graphql-codegen/typescript": "^1.3.1",
      "@graphql-codegen/typescript-resolvers": "^1.3.1",
      "ts-node": "^8.3.0",
      "typescript": "^3.5.2"
   }
```

Module call:
```js
import 'reflect-metadata';
// see: https://graphql-modules.com/docs/recipes/development-environment#import-from-graphql-files
import 'graphql-import-node';
import { GraphQLModule } from '@graphql-modules/core';
import { UserModule } from './user/user.module';

export const AppModule = new GraphQLModule({
  imports: [
    // AuthModule, UserModule, SocialNetworkModule
    UserModule.forRoot({
      truc: 'tree',
      secretKey: '123',
      remoteEndpoint: 'http://my-other-service.com',
    }),
  ]
});
```

Module définition
```js
import { gql } from 'apollo-server';
import * as typeDefs from './schema.graphql';
import resolvers from './resolvers';
import { UserProvider } from './provider';
import { GraphQLModule } from '@graphql-modules/core';

export interface UserModuleConfig {
  truc: string;
  secretKey: string;
  remoteEndpoint: string;
}

export const UserModule = new GraphQLModule<UserModuleConfig>({    
  name: 'user-module',  // without name DI not working
  typeDefs,
  resolvers,
  providers: ()=> [
      UserProvider,
  ],  
});
```

And provider:
```js
@Injectable()
export class UserProvider {

  private users: User[] = [{ id: '0', username: 'jhon' }];

  // not working
  //constructor(@Inject(ModuleConfig( (module) => UserModule)) private config: UserModuleConfig) {
  
  // working
  constructor(@Inject(ModuleConfig('user-module')) private config: UserModuleConfig) {
    logger.info('[UserModule] injection config for remoteEndpoint:' + this.config.remoteEndpoint);
  }

  public getUserById(id: string): User | undefined {
    return this.users.find( (user) => user.id === id);
  }

  public onInit(): void {    
    logger.info('[UserModule] onInit');
  }
}
```

When i using ```ModuleConfig( module => UserModule) ``` insteed  ```ModuleConfig('user-module')) ``` an exception occured:
```
GraphQL-Modules Error: Dependency Provider Not Found!
      - Provider #Symbol(ModuleConfig.undefined) couldn't be injected into Provider #UserProvider
      -- Provider #Symbol(ModuleConfig.undefined) is not provided in #Module user-module scope!

      Possible solutions:
      - Make sure you have imported the module of Provider #Symbol(ModuleConfig.undefined) in the module of Provider #UserProvider
```

Regards

**Comment by ardatan** [comment]
You should do `@Inject(ModuleConfig(module => MyModule)` not `@Inject(ModuleConfig(MyModule)`.

**Comment by ardatan** [comment]
I think it is related to circular imports so could you try to inject like below with the following version;
`@Inject(ModuleConfig) private config: UserModuleConfig`
`0.7.8-alpha-ce9645b.27+ce9645b`

**Comment by ardatan** [comment]
Fixed in 0.7.8!
